staging/src/k8s.io/apiserver/pkg/server/options/admission.go

```diff
@@ -143,9 +143,8 @@ func (a *AdmissionOptions) ApplyTo(
 	if err != nil {
 		return err
 	}
-	genericInitializer := initializer.New(clientset, informers, c.Authorization.Authorizer, features)
-	initializersChain := admission.PluginInitializers{}
-	pluginInitializers = append(pluginInitializers, genericInitializer)
+	genericInitializer := initializer.New(clientset, informers, c.Authorization.Authorizer, features, c.DrainedNotify())
+	initializersChain := admission.PluginInitializers{genericInitializer}
```

Comment:
This is on purpose changing the order of plugin initializers.

Previously the one setting QuotaEvaluator:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubeapiserver/admission/initializer.go#L64
was called before the genericInitializer that was setting stopCh added in this PR.

Reply:
that's really subtle... how do we know this isn't breaking dependencies in the opposite direction?

Reply:
I was looking into it and i didn't find anything suspicious.

The alternatively would be to adjust resource quota to also set stopCh of the underlying QuotaEvaluator, which fixes this particular problem, but doesn't address a potential future problems...

Reply:
> (and ensure new reverse order dependencies don't get added)

The check that I added in resource quota that stopCh is actually set properly ensure that if this order will get changed, the initialization of resource quota plugin will start failing.
In general, every plugin that needs that, will have to handle it on their own.